### ISSUE TITLE
NPM warning on install: postcss-reporter@4.0.0 requires a peer of postcss@^6.0.0 but none was installed

### ISSUE THREAD

**Issue by sinedied** [stylelint/stylelint#2779]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

When installing `stylelint` with `npm@4` a warning is issued: 
```bash
npm WARN postcss-reporter@4.0.0 requires a peer of postcss@^6.0.0 but none was installed.
 ```
```
npm -v
4.6.1
node -v
v7.9.0
```

> Which rule, if any, is this issue related to?

N/A

> What CSS is needed to reproduce this issue?

N/A
> What stylelint configuration is needed to reproduce this issue?

N/A

> Which version of stylelint are you using?

`8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint "src/**/*.scss" --syntax scss`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

N/A

> What did you expect to happen?

Install without warnings.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were issued with NPM (does not happen with `yarn`)

```bash
npm WARN postcss-reporter@4.0.0 requires a peer of postcss@^6.0.0 but none was installed.
 ```

**Comment by ntwb** [comment]
Please update npm to the latest 5.x release, it is fixed with npm 5

Previously with npm 3.x and 4.x was and idiosyncratic issue with how npm installs modules, if you were to run `npm install postcss-reporter postcss` then this warning would not appear.

Aside: You should also update from Node.js 7.x to the latest 8.x

**Comment by sinedied** [comment]
@ntwb Same issue occurs with `node@8.2.1` and `npm@5.3.0`, see this log on Travis: https://travis-ci.org/ngx-rocket/generator-ngx-rocket/jobs/256998435

**Comment by gussyboy** [comment]
@ntwb you're solution isn't ideal for people who are using the LTS version of node, which ships with npm 3.x

I guess for the time being we'll just have to install `postcss` as a package

**Comment by hudochenkov** [comment]
@gussyboy npm is not tied to Node version, and `npm@5` could be installed and used on Node 4+.

**Comment by Arcanemagus** [comment]
@hudochenkov @gussyboy Not only _can_ NPM be installed on Node.js 4+, as far as the NPM team is concerned _anything_ other than the latest NPM version is outdated software and shouldn't be used.

**Comment by ntwb** [comment]
The latest npm 5.x should be installed and used for all Node.js releases. npm clarified this recently in the following blog post http://blog.npmjs.org/post/162451604515/the-npm-clis-long-term-support-lts-policy

**Comment by ntwb** [comment]
To the core of the issue here at hand though:

https://github.com/postcss/postcss-reporter/blob/master/package.json#L42-L44

This is an issue with the `postcss-reporter` module and not stylelint.

The state of `peerDependencies` is ¯\\_(ツ)_/¯ for me at the moment, I believe they were going to be deprecated, but not entirely sure and can't find a reference to that right now.

Closing this as not a stylelint issue

**Comment by sinedied** [comment]
@ntwb I reported the issue on `postcss-reporter`, but the real issue still lies here with stylelint in my opinion: if you decide to include package `X` and package `X` says it needs package `Y` to be installed along, your dependencies should be both on `X` and `Y`.

To solve this, `stylelint` would need to add a dependency on `postcss`, since you use `postcss-reporter`.

**Comment by hudochenkov** [comment]
@sinedied we [have](https://github.com/stylelint/stylelint/blob/55bfb4d81db96e637f827d7d87fdbdccec230615/package.json#L61-L64) `postcss@6` as a dependency. It's some npm bug.

**Comment by sinedied** [comment]
Ouch :/ But apart from using `yarn`, how can I resolve this since using `npm@5` does not solve the issue for the CI?

**Comment by hudochenkov** [comment]
@sinedied it's just a warning. It's annoying, but at least it doesn't break a build.

**Comment by sinedied** [comment]
@hudochenkov actually, it does! When using Cordova, if npm emits warning or errors it breaks the cordova build :/

**Comment by Arcanemagus** [comment]
> The state of `peerDependencies` is ¯\(ツ)/¯ for me at the moment, I believe they were going to be deprecated, but not entirely sure and can't find a reference to that right now.

@ntwb Maybe you were thinking of the `prepublish` changes? `peerDependencies` is a _critical_ part of many package's ecosystem (including `stylelint`!) so I highly doubt it's going anywhere.

**Comment by sinedied** [comment]
@ntwb an update of `postcss-reporter` was pushed that changed `postcss` as a dependency, not a peer dependency. Can you reflect the change in `stylelint` to fix this nom issue?

**Comment by sinedied** [comment]
@ntwb bump! Could you please update to the latest `postcss-reporter` version that fix the issue?

The warning can be reproduced with `npm@5`, and it breaks cordova builds 😞

**Comment by ntwb** [comment]
@sinedied No need for a PR, `postcss-reporter` was updated to `5.0.0` in #2786

Per above, we'll look to releasing stylelint 8.1.0 that contains this change in the coming days

**Comment by EldinZaimovic** [comment]
I have this problem again with postcss 7.0.0
```
npm WARN @stylelint/postcss-css-in-js@0.37.1 requires a peer of postcss@>=7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @stylelint/postcss-markdown@0.36.1 requires a peer of postcss@>=7.0.0 but none is installed. You must install peer dependencies yourself.
```
